src/qfvm/circuit.hpp

```diff
@@ -100,6 +101,18 @@ void Circuit::add_op(QuantumOperator &op){
     }
 }
 
+vector<QuantumOperator> Circuit::gates(){
```

Comment:
where's this function called

Reply:
The gpu simulator will call: https://github.com/ScQ-Cloud/pyquafu/blob/93309771a5b7465416ec20ef9dbdfd6b5ad2ce2f/src/qfvm_gpu/cuda_simulator.cuh#L26C39-L26C39